blocks/log.py

```diff
@@ -64,10 +69,3 @@ def previous_row(self):
     @property
     def last_epoch_row(self):
         return self[self.status['_epoch_ends'][-1]]
-
-    def to_dataframe(self):
-        """Convert a log into a :class:`.DataFrame`."""
-        if not PANDAS_AVAILABLE:
-            raise ImportError("The pandas library is not found. You can"
-                              " install it with pip.")
-        return DataFrame.from_dict(self, orient='index')
```

Comment:
I generally write my own quick helper function, something like
```python
def get_values(v):
    return [(t, log[t][v]) for t in log if v in log[t]]
plt.plot(*zip(*get_values('valid_error')))
```
Not everyone uses Pandas all the time. I know it's powerful, but I personally find it pretty unwieldy, so I don't use it unless I need to.

As I mentioned in the other comment, I'm personally not a fun of bits and pieces of the library not working even after the user installed everything by the book. It makes the code messier (import-traps everywhere, custom warnings in the code) and catches users by surprise. I think it's a valid thing to do if we really need a big library that we know we won't use anywhere else, but if it's a small library I would suggest just adding it (e.g. `progressbar2`) and if it's a big library that can easily be avoided, I'd suggest getting rid of it entirely (e.g. `pandas`).